core/storage/suggestion/gae_models.py

```diff
@@ -700,6 +702,17 @@ def get_in_review_translation_suggestions_with_exp_ids_by_offset(
                     given exploration IDs.
                 next_offset: int. The input offset + the number of results
                     returned by the current query.
+
+        Note:
```

Comment:
@seanlip, In regards to our discussion and your prev comment, I decided to add a Note. PTAL and let me know if it can serve the purpose.

Reply:
The content of this note is good, but please put it in the arg's docstring. We don't have support for "Note:" sections in docstrings (at the same level as Args/Returns/Raises) and we don't want to introduce widespread docstring format changes due to just this case.

The description could be something like this:

```
Maximum number of entities to be returned; if None, returns all matching entities. Note that the None case is needed because we sometimes need to fetch all translation suggestions so that they can be sorted client-side based on specific criteria, and the datastore does not efficiently support sorting based on complex client-side criteria. So, we need to fetch all matching entities from the datastore and allow the frontend to perform the sorting as needed. Even in this case, though, the maximum number of entities that will be returned by this query is XXX.
```

This also raises one question. When writing this docstring, I felt I needed to put something in at the end to talk about bounding. If we need data for all cards within ONE exploration, then that is capped by the max number of cards in an exploration, which is typically 300 at most (so 500 is a safe limit). However, the interface for this function supports multiple exploration IDs. Do we need to ever fetch data without limit for multiple IDs? If so, why? If not, I suggest you have a separate method that is scoped to a single exploration ID and not support a limit for that method (since you can explain that it's bounded above anyway), and keep the limit for the current method required.

Reply:
Currently the fetch is made in one other place here: https://github.com/oppia/oppia/blob/develop/core/controllers/contributor_dashboard.py#L387. that fetches data without limit for multiple IDs in order to get all the reviewable suggestions for a particular topic.

Reply:
Thanks @sagangwee.  What do we use this data for (why do we need all reviewable suggestions for a particular topic)? From the UI perspective, when the topic is selected in the UI then I see a bunch of lessons; when I click Review on the lesson then I see its cards. At no point would the code actually need to get all suggestions for a topic at once... it would need to do an initial fetch of 1 opportunity per lesson to see whether to show the lesson or not, and then a fetch for all the open opportunities within a specific lesson.

I think it's fine for this function to have no limit if it's only going to ever be used for one lesson, but I think that setting up this API to handle multiple exploration IDs and have no limit is an invitation for it to break in the future. If the existing usage should be updated based on this observation then I would suggest making a new method for the 1-exploration case and adding a TODO to transition the existing callsite to use that instead (and deprecate the no-limit method).

Reply:
Got it. Agree with the TODO suggestion.

Reply:
I have made a new method for 1-exploration case and added a TODO, PTAL.